homeassistant/components/sensor/torque.py

```diff
@@ -0,0 +1,117 @@
+"""
+homeassistant.components.sensor.torque
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Get data from the Torque OBD application.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.torque/
+"""
+
+import re
+from homeassistant.const import HTTP_OK
+from homeassistant.helpers.entity import Entity
+
+
+DOMAIN = 'torque'
+DEPENDENCIES = ['http']
+SENSOR_EMAIL_FIELD = 'eml'
+DEFAULT_NAME = 'vehicle'
+ENTITY_NAME_FORMAT = '{0} {1}'
+
+API_PATH = '/api/torque'
+SENSOR_NAME_KEY = r'userFullName(\w+)'
+SENSOR_UNIT_KEY = r'userUnit(\w+)'
+SENSOR_VALUE_KEY = r'k(\w+)'
+
+NAME_KEY = re.compile(SENSOR_NAME_KEY)
+UNIT_KEY = re.compile(SENSOR_UNIT_KEY)
+VALUE_KEY = re.compile(SENSOR_VALUE_KEY)
+
+
+def decode(value):
+    """ Double-decode required. """
+    return value.encode('raw_unicode_escape').decode('utf-8')
+
+
+def convert_pid(value):
+    """ Convert pid from hex string to integer. """
+    return int(value, 16)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Set up Torque platform. """
+
+    vehicle = config.get('name', DEFAULT_NAME)
+    email = config.get('email', None)
+    sensors = {}
+
+    def _receive_data(handler, path_match, data):
+        """ Received data from Torque. """
+        handler.send_response(HTTP_OK)
+        handler.end_headers()
+
+        if email is not None and email != data[SENSOR_EMAIL_FIELD]:
```

Comment:
I am not convinced yet about email filtering but I'm fine with it for MVP. Just so you know, Home Assistant will only trigger the first HTTP endpoint that matches the url. So loading this component twice will not be possible.